src/Model/Search.php

```diff
@@ -29,4 +39,207 @@ public static function getUserTags()
+		    (substr(Strings::normaliseLink($user), 0, 7) == "http://")) {
+
+			$user_data = Probe::uri($user);
+			if (empty($user_data)) {
+				return null;
+			}
+
+			if (!(in_array($user_data["network"], [Protocol::ACTIVITYPUB, Protocol::DFRN, Protocol::OSTATUS, Protocol::DIASPORA]))) {
+				return null;
+			}
+
+			$contactDetails = Contact::getDetailsByURL(defaults($user_data, 'url', ''), local_user());
+			$itemurl = (($contactDetails["addr"] != "") ? $contactDetails["addr"] : defaults($user_data, 'url', ''));
+
+			$result = new Result(
+				defaults($user_data, 'name', ''),
+				defaults($user_data, 'addr', ''),
+				$itemurl,
+				defaults($user_data, 'url', ''),
+				defaults($user_data, 'photo', ''),
+				defaults($user_data, 'network', ''),
+				defaults($contactDetails, 'cid', 0),
+				0,
+				defaults($user_data, 'tags', '')
+			);
+
+			return new ResultList(1, 1, 1, [$result]);
+
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * Search in the global directory for occurrences of the search string
+	 * This is mainly based on the JSON results of https://dir.friendica.social
+	 *
+	 * @param string $search
+	 * @param int    $page
+	 *
+	 * @return ResultList|null
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	public static function searchDirectory($search, $page = 1)
+	{
+		$config = self::getApp()->getConfig();
+		$server = $config->get('system', 'directory', self::DEFAULT_DIRECTORY);
+
+		$searchUrl = $server . '/search?q=' . urlencode($search);
+
+		if ($page > 1) {
+			$searchUrl .= '&page=' . $page;
+		}
+
+		$red = 0;
+		$resultJson = Network::fetchUrl($searchUrl, false,$red, 0, 'application/json');
+
+		$results    = json_decode($resultJson, true);
+
+		$resultList = new ResultList(
+			defaults($results, 'page', 1),
+			defaults($results, 'count', 1),
+			defaults($results, 'itemsperpage', 1)
+		);
+
+		$profiles = defaults($results, 'profiles', []);
+
+		foreach ($profiles as $profile) {
+			$contactDetails = Contact::getDetailsByURL(defaults($profile, 'profile_url', ''), local_user());
+			$itemurl = (!empty($contactDetails['addr']) ? $contactDetails['addr'] : defaults($profile, 'profile_url', ''));
+
+			$result = new Result(
+				defaults($profile, 'name', ''),
+				defaults($profile, 'addr', ''),
+				$itemurl,
+				defaults($profile, 'profile_url', ''),
+				defaults($profile, 'photo', ''),
+				Protocol::DFRN,
+				defaults($contactDetails, 'cid', 0),
+				0,
+				defaults($profile, 'tags', ''));
+
+			$resultList->addResult($result);
+		}
+
+		return $resultList;
+	}
+
+	/**
+	 * Search in the local database for occurrences of the search string
+	 *
+	 * @param string $search
+	 * @param int    $start
+	 * @param int    $itemPage
+	 * @param bool   $community
+	 *
+	 * @return ResultList|null
+	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
+	 */
+	public static function searchLocal($search, $start = 0, $itemPage = 80, $community = false)
```

Comment:
Same gripe about a possible misplacement of this method. Maybe `Core\Search::getContactsFromLocalDirectory`?